### PR TITLE
Adds stathat logging to permalink signup

### LINKED ISSUE #6075: Monitoring: Sign Up Via Permalink
#### Experience to Test:

Signing up via permalink.
#### Proxy:

No POST requests occur.
#### System:

StatHat

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?

Adds stat hat logging to permalink signups
#### How should this be manually tested?

If you do a normal signup, does this log?
If you do a permalink signup, does this log?
#### Any background context you want to provide?

I couldn't see a better way than using the $source parameter / URL to figure out if this was the perma link so just went with that. 
#### What are the relevant tickets?

Fixes #6075

### CODE REVIEW

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.module [hunk 1]
```diff
@@ -243,6 +243,10 @@ function dosomething_signup_create($nid, $uid = NULL, $source = NULL, $timestamp
     return FALSE;
   }
 
+  if (strpos($source, 'reportback') === 0) {
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: `if (module_exists('stathat')) {`?

Comment (angaither) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: why not just if `$source === 'reportback'`?

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: @sergii-tkachenko ah good catch, we have a few places this needs to be updated then https://github.com/DoSomething/phoenix/blob/a47d9650be54a127adaa5e80f297c474f431fc01/lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module#L96

@angaither the entire source param is like reportback/3423/djsfu4w89rd9 just the start of it is reportback

Comment (angaither) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: @joe the line above checks for stathat. https://github.com/DoSomething/phoenix/blob/a47d9650be54a127adaa5e80f297c474f431fc01/lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module#L95

got it, i forgot we kept which rb got the person to sign up

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: ahh missed that

**Comment by itsjoekent** [comment]
@sergii-tkachenko alrighty should be good to go

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.module (hunk 1 above)
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: @angaither @deadlybutter 

> why not just if `$source === 'reportback'`
> To express all sources that start with reportback, ex. `reportback-foobar`?

**Comment by angaither** [comment]
@deadlybutter this should be at the end of the signup_create method, so we know it actually succeeded

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.module [hunk 2]
```diff
@@ -243,6 +243,13 @@ function dosomething_signup_create($nid, $uid = NULL, $source = NULL, $timestamp
     return FALSE;
   }
 
+  if (module_exists('stathat')) {
+    // The source is often a string longer than just "reportback", this logic accounts for that.
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: > The source is often a string longer than just "reportback"

How longer? Should it better clarify that the source string starts with `reportback`?

**Comment by itsjoekent** [comment]
@angaither moved it right before the return entity->sid

@sergii-tkachenko updated comment

**Comment by sergiitk** [comment]
:+1:

**Pull request merged**